homeassistant/components/frank_energie/config_flow.py

```diff
@@ -0,0 +1,86 @@
+"""Config flow for Picnic integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from python_frank_energie import FrankEnergie
+from python_frank_energie.exceptions import AuthException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_AUTHENTICATION, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONF_AUTH_TOKEN, CONF_REFRESH_TOKEN, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the config flow for Frank Energie."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+
+    async def async_step_login(self, user_input=None, errors=None) -> FlowResult:
+        """Handle login with credentials by user."""
+        if not user_input:
+            data_schema = vol.Schema(
+                {
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            )
+
+            return self.async_show_form(
+                step_id="login",
+                data_schema=data_schema,
+                errors=errors,
+            )
+
+        async with FrankEnergie() as api:
+            try:
+                auth = await api.login(
+                    user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+                )
+            except AuthException as ex:
+                _LOGGER.exception("Error during login", exc_info=ex)
+                return await self.async_step_login(errors={"base": "invalid_auth"})
```

Comment:
Oh right, I now see why you had errors in the parameters. Suggestion: maybe you can turn this function around
```py
errors: dict[str, str] = {}
if user_input:
    # logic to verify it works and create entry, else we set errors
return self.async_show_form(........, errors=errors)

Reply:
Done :)